minifi-nar-bundles/minifi-framework-bundle/minifi-framework/minifi-runtime/src/main/java/org/apache/nifi/minifi/configuration/FileChangeNotifier.java

```diff
@@ -42,33 +43,18 @@
  */
 public class FileChangeNotifier implements Runnable, ConfigurationChangeNotifier, Closeable {
 
+    private Path configFile;
+    private WatchService watchService;
+    private long pollingSeconds;
+
     private final ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
-    private final Path configFile;
-    private final WatchService watchService;
     private final Set<ConfigurationChangeListener> configurationChangeListeners = new HashSet<>();
 
-    private static final int DEFAULT_POLLING_PERIOD_INTERVAL = 15;
-    private static final TimeUnit DEFAULT_POLLING_PERIOD_UNIT = TimeUnit.SECONDS;
-
-    /**
-     * @param configFile to monitor for changes
-     * @throws IOException if there are any issues with accessing the specified config file or generating the associated {@link WatchService}.
-     */
-    public FileChangeNotifier(Path configFile) throws IOException {
-        this(configFile, initializeWatcher(configFile));
-    }
-
-
-    public FileChangeNotifier(Path configFile, WatchService watchService) {
-        final File file = configFile.toFile();
-        if (!file.exists() || !file.canRead() || !file.isFile()) {
-            throw new IllegalArgumentException(String.format("The specified path %s must be a readable file.", configFile));
-        }
-
-        this.configFile = configFile;
-        this.watchService = watchService;
-    }
+    protected static final String CONFIG_FILE_PATH_KEY = "nifi.notifier.file.config.path";
+    protected static final String POLLING_PERIOD_INTERVAL_KEY = "nifi.notifier.file.polling.period.seconds";
```

Comment:
Should include "minifi" either by replacing "nifi" with it or by adding it subsequently.